src/common/util/include/openvino/util/file_util.hpp

```diff
@@ -355,6 +355,24 @@ void save_binary(const std::string& path, const char* binary, size_t bin_size);
  */
 const char* trim_file_name(const char* const fname);
 
+#ifndef __EMSCRIPTEN__
+/**
+ * @brief Check the file path is symlink or hardlink
+ * @param path - file path to be checked
+ * @return Return true, if the path is symlink or hardlink
+ */
+bool is_symlink_or_hardlink(const std::string& path);
```

Comment:
It can be just alias to std function depends on pre-processor option.

For not supported it could `static constexpr` always returns true or false. Then some of #ifdef can be removed from .cpp and code will be optimized as condition will compile time value.